### PR TITLE
feat: import cache

### PR DESCRIPTION
**Pull request by fsuits**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

Import now uses cache, and the cache system is generalized to handle relative paths.

### CODE REVIEW

File: tests/trestle/core/remote/cache_test.py [hunk 1]
```diff
@@ -41,33 +42,42 @@ def as_file_uri(path: str) -> str:
...
-    fetcher = cache.FetcherFactory.get_fetcher(pathlib.Path(tmp_trestle_dir), catalog_file)
-    # Create/update the cache copy
+    catalog_data.oscal_write(catalog_file)
+    saved_data = fs.load_file(pathlib.Path(catalog_file))
+    if relative:
+        catalog_str = f'./catalogs/{cat_name}' if in_trestle else f'../{cat_name}'
+    else:
+        catalog_str = str(catalog_file)
+    fetcher = cache.FetcherFactory.get_fetcher(tmp_trestle_dir, catalog_str)
+    return fetcher, catalog_data, saved_data
```
Comment (butler54) [inline tests/trestle/core/remote/cache_test.py:65]: I'm not sure why we are saving then loading data here. The extra read seems unnecessary (even if we need the file). Users can ask for the dict data by  OscalModel.dict(by_alias=True, exclude_unset=True)

File: trestle/core/remote/cache.py [hunk 2]
```diff
@@ -110,11 +110,16 @@ def get_raw(self, force_update=False) -> Dict[str, Any]:
...
         try:
-            return fs.load_file(self._cached_object_path)
-        except Exception as e:
-            logger.error(f'Cannot fs.load_file {self._cached_object_path}')
-            logger.debug(e)
-            raise TrestleError(f'Cache get failure for {self._uri}') from e
+            raw_data = fs.load_file(self._cached_object_path)
+        except Exception:
+            try:
+                # files from NIST seem to need cp1252 encoding
```
Comment (butler54) [inline trestle/core/remote/cache.py:116]: NIST files are annotated with utf-8 as the extension type

**Review by butler54 — APPROVED** [review]
LGTM see the small changes.

File: tests/trestle/core/remote/cache_test.py (hunk 1 above)
Comment (fsuits) [reply tests/trestle/core/remote/cache_test.py:65]: This was needed in order to test the raw mode of fetching - but I have removed the extra read and the corresponding test of raw read.

File: trestle/core/remote/cache.py (hunk 2 above)
Comment (fsuits) [reply trestle/core/remote/cache.py:116]: Oh - that comment should have been removed.  At first I thought NIST had an issue, but the issue was in cache.py where it wrote the file with pathlib.write_text().  That line now has encoding specified and before it did not - causing the problem.

Comment now removed.

**Pull request merged**